core/src/main/java/io/undertow/server/HttpServerExchange.java

```diff
@@ -165,7 +166,7 @@ public final class HttpServerExchange extends AbstractAttachable {
 
     // mutable state
 
-    private int state = 200;
+    private AtomicInteger state = new AtomicInteger(200);
```

Comment:
@burov4j  Thanks for your PR!
I would like to follow the same pattern we have in XNIO. Could you instead make the field volatile and add an atomic field updater, like in this class: https://github.com/xnio/xnio/blob/3.x/api/src/main/java/org/xnio/Connection.java ?
This will have a better memory footprint than the currently proposed solution.

Reply:
Thanks @burov4j !